pkg/kv/kvserver/replica_sst_snapshot_storage.go

```diff
@@ -87,6 +126,9 @@ func (s *SSTSnapshotStorageScratch) createDir() error {
 func (s *SSTSnapshotStorageScratch) NewFile(
 	ctx context.Context, bytesPerSync int64,
 ) (*SSTSnapshotStorageFile, error) {
+	if s.closed {
+		panic("closed")
```

Comment:
nit: since we have an error return path here, let's return an `AssertionFailedf` rather than panic'ing. Ditto elsewhere.